SourceCode/raytracer/Renderer.h

```diff
@@ -12,12 +12,92 @@ class Renderer {
 
     void render(Buffer2D& image) const
     {
-
+        for (int y = 0; y < camera.getHeight(); ++y) {
+            for (int x = 0; x < camera.getWidth(); ++x) {
+                Ray ray = camera.generateRay(x, y);
+                // Color color = traceImagePlane(ray); // hw3
+                Color color = traceRay(ray); // hw5
+                image(x, y) = color;
+            }
+        }
     }
 
 private:
     const Camera& camera;
     // const Scene& scene; // TODO: Implement Scene class
 
+    Color traceRay(const Ray& ray) const
+    {
+        
+        // alls vertices even with the image plane
+        // This one does not cause problems
+        Triangle tri_even
+        {
+            Vec3{-1.f, -1.f, -1.f},
+            Vec3{1.f, -1.f, -1.f},
+            Vec3{0.f, 1.f, -1.f},
+        };
+
+        // Bottom vert deeper
+        Triangle tri_bot
+        {
+            Vec3{-1.f, -1.f, -1.f},
+            Vec3{1.f, -1.f, -1.f},
+            Vec3{0.f, 1.f, -1.5f}
+        };
+
+        // right vert deeper
+        Triangle tri_right
+        {
+            Vec3{-1.f, 1.f, -1.f},
+            Vec3{-1.f, -1.f, -1.f},
+            Vec3{1.f, 0.f, -1.5f},
+        };
+
+        // top vert deeper
+        Triangle tri_top
+        {
+            Vec3{-1.f,1.f,-1.f},
+            Vec3{0.f,1.f,-1.5f},
```

Comment:
And here, the Y component should probably be -1.f (you still get a triangle this way, but from the perspective of the camera it will be very skewed).